libxmf/XmfVpxDecoder.c

```diff
@@ -0,0 +1,106 @@
+#include "XmfVpxDecoder.h"
+#include <stdlib.h>
+#include <string.h>
+#include <vpx/vp8dx.h>
+#include <vpx/vpx_decoder.h>
+#include "XmfVpxImage.h"
+
+// Internal decoder structure
+struct xmf_vpx_decoder
+{
+    vpx_codec_ctx_t codec;
+    XmfVpxDecoderError lastError;
+};
+
+XmfVpXDecoder *XmfVpxDecoder_Create(XmfVpxDecoderConfig cfg)
+{
+
+    XmfVpXDecoder *decoder = (XmfVpXDecoder *)malloc(sizeof(XmfVpXDecoder));
+    if (!decoder)
+    {
+        return NULL;
+    }
+
+    memset(decoder, 0, sizeof(XmfVpXDecoder));
+    decoder->lastError.code = NO_ERROR;
+
+    vpx_codec_dec_cfg_t dec_cfg = {
+        .threads = cfg.threads,
+        .w = cfg.w,
+        .h = cfg.h};
+
+    vpx_codec_iface_t *iface = (cfg.codec == VP8) ? vpx_codec_vp8_dx() : vpx_codec_vp9_dx();
+
+    vpx_codec_err_t res = vpx_codec_dec_init(&decoder->codec, iface, &dec_cfg, 0);
+    if (res != VPX_CODEC_OK)
+    {
+        decoder->lastError.code = INIT_ERROR;
```

Comment:
Pointless to do this if you immediately free the decoder and return NULL

Reply:
updated